### PR TITLE
dex-1329 - show collaboration type in table

### PR DESCRIPTION
**Pull request by Emily-Ke**
Pull request checklist:

- [x] Features and bugfixes should be PRed into the `develop` branch, **not** `main`
- [x] All text is internationalized
- [x] There are no linter errors
- [x] New features support all states (loading, error, etc)

Which JIRA ticket(s) and/or GitHub issues does this PR address?
  - DEX-1329 - Manage Users: see whether collaboration is edit or view only

---

## Pull Request Overview

- Changes what is displayed in the user manager collaborations table.
- Adds a dialog for user managers to edit a collaboration.
dialog

**Review Notes**
- This code requires WildMeOrg/houston#795

### CODE REVIEW

**Comment by Emily-Ke** [comment]
This code requires WildMeOrg/houston#795

File: src/models/collaboration/model.js
Comment (brmscheiner) [inline src/models/collaboration/model.js:1]: seems like the name of this file should be `utils.js`. otherwise this is really nice

File: src/pages/userManagement/components/EditCollaborationDialog.jsx
Comment (brmscheiner) [inline src/pages/userManagement/components/EditCollaborationDialog.jsx:44]: can you use one of the userManagement utils in the file below? seems like this logic is getting repeated in a lot of different places in the app (not just this PR). i think we should lift this logic to the top of the app and apply it the same everywhere

File: src/pages/userManagement/components/EditCollaborationDialog.jsx
Comment (brmscheiner) [inline src/pages/userManagement/components/EditCollaborationDialog.jsx:51]: switch statements can be really prone to bugginess. this one looks okay but in general i prefer if statements (even if it doesn't look quite as good)

File: src/pages/userManagement/components/EditCollaborationDialog.jsx
Comment (brmscheiner) [inline src/pages/userManagement/components/EditCollaborationDialog.jsx:130]: these comments arent really necessary

File: src/pages/userManagement/components/EditCollaborationDialog.jsx
Comment (brmscheiner) [inline src/pages/userManagement/components/EditCollaborationDialog.jsx:149]: `style={{ flexGrow: 1, maxWidth: index === 1 ? 280 : 'unset' }}`
 (if you like it better, not a huge deal either way)

File: src/pages/userManagement/components/EditCollaborationDialog.jsx
Comment (brmscheiner) [reply src/pages/userManagement/components/EditCollaborationDialog.jsx:44]: (though you dont need to do that to merge this PR)

**Review by brmscheiner — CHANGES_REQUESTED** [review]

**Review by brmscheiner — APPROVED** [review]
sorry, clicked the wrong button!

File: src/models/collaboration/model.js
Comment (Emily-Ke) [reply src/models/collaboration/model.js]: moved to `collaborationUtils.js` in https://github.com/WildMeOrg/codex-frontend/pull/451/commits/62104e7a6d783092e520aba2ad9bd9b3ee349b91.

File: src/pages/userManagement/components/EditCollaborationDialog.jsx
Comment (Emily-Ke) [reply src/pages/userManagement/components/EditCollaborationDialog.jsx]: Changed to `if` statements in https://github.com/WildMeOrg/codex-frontend/pull/451/commits/466d459bf84ef3ece5a0152ab78512b40070529c.

File: src/pages/userManagement/components/EditCollaborationDialog.jsx
Comment (Emily-Ke) [reply src/pages/userManagement/components/EditCollaborationDialog.jsx]: Removed in https://github.com/WildMeOrg/codex-frontend/pull/451/commits/0e0918d3e8d4c5901242ff18c8f4edecf9b494b3.

File: src/pages/userManagement/components/EditCollaborationDialog.jsx
Comment (Emily-Ke) [reply src/pages/userManagement/components/EditCollaborationDialog.jsx]: Thanks. I changed it in https://github.com/WildMeOrg/codex-frontend/pull/451/commits/0e0918d3e8d4c5901242ff18c8f4edecf9b494b3.

File: src/pages/userManagement/components/EditCollaborationDialog.jsx
Comment (Emily-Ke) [reply src/pages/userManagement/components/EditCollaborationDialog.jsx]: I made some changes. Let me know what you think:
1. There was already a FormattedReporter component from PR #444, so I updated AssetGroup and SightingEntityHeader to use it. (https://github.com/WildMeOrg/codex-frontend/pull/451/commits/2205d20605011b8c9d6d676e1ddec4d4c973bcb0)
2. I applied the `formatUserMessage` from this PR, which includes the email address in the result, to the collaboration management form. (https://github.com/WildMeOrg/codex-frontend/pull/451/commits/269a252bdbd35360a2632f1b0a719c5df34eb773)
    - I left `formatUserMessage` in the `pages/userManagement` directory because I think the name and email combination should only be used for user management right now. For some reason the `collaborationManagementForm` is not in that directory, but I think it should probably be in `userManagement/components`. I'll move it if you agree.
3. I removed this `FormattedUserName` component and instead crated a utility function in the general formatters file, also named `formatUserMessage` that just returns either the user's name or Unnamed User. (https://github.com/WildMeOrg/codex-frontend/pull/451/commits/94f708d01372f2cbfae95319e2368ba13277ebfd)

- I named both functions `formatUserMessage` because that is what they are doing. The user management one just needs more information. I could rename one, or expand the general formatter to handle including or ignoring the email address. I don't really want to put too much in one function right now because we seem to have a few similar, but slightly different needs for almost the same text.
- Within the user formatter functions and components, I used `fullName` because that is our convention for identifiers. However, because Houston sends `full_name` that means that in a lot, although not all, of the places the function is called, instead of just calling it with something like `formatUserMessage(user)`, we have to call it with `formatUserMessage({ fullName: user?.full_name })`.

Comment (brmscheiner) [reply src/pages/userManagement/components/EditCollaborationDialog.jsx]: 1. great!
2. great! i agree that `collaborationManagementForm` should be moved.
3. this is probably fine for now... eventually it could be good to build a single function that can accept a variety of different inputs but always tries to return a formatted user message. we can talk more about this later but i think it's a pattern we could use in a lot of places

**Review by brmscheiner — APPROVED** [review]

**Comment by Emily-Ke** [comment]
🛑 DO NOT MERGE UNTIL WildMeOrg/houston#795 is merged.

**Pull request merged**
